packages/bigint-uint8array/src/converter/native.ts

```diff
@@ -1,17 +1,29 @@
-import * as native from 'bigint-buffer'
-import { type UncheckedConverter } from './type'
+import {type UncheckedConverter} from './type'
+import {uncheckedConverter as fallback} from './browser.js';
+let _uncheckedConverter: UncheckedConverter = fallback;
 
-export const uncheckedConverter: UncheckedConverter = {
-  bigEndianToNewArray: native.toBufferBE,
-  bigEndianToArray: (num: bigint, dest: Uint8Array) => {
-    const src = uncheckedConverter.bigEndianToNewArray(num, dest.length)
-    dest.set(src)
-  },
-  littleEndianToNewArray: native.toBufferLE,
-  littleEndianToArray: (num: bigint, dest: Uint8Array) => {
-    const src = uncheckedConverter.littleEndianToNewArray(num, dest.length)
-    dest.set(src)
-  },
-  arrayToLittleEndian: native.toBigIntLE as UncheckedConverter['arrayToLittleEndian'],
-  arrayToBigEndian: native.toBigIntBE as UncheckedConverter['arrayToBigEndian']
+try {
+  // eslint-disable-next-line @typescript-eslint/no-var-requires
+  const module = await import('bigint-buffer');
+
+  if (module) {
+    _uncheckedConverter = {
+      bigEndianToNewArray: module.toBufferBE,
+      bigEndianToArray: (num: bigint, dest: Uint8Array) => {
+        const src = _uncheckedConverter.bigEndianToNewArray(num, dest.length)
+        dest.set(src)
+      },
+      littleEndianToNewArray: module.toBufferLE,
+      littleEndianToArray: (num: bigint, dest: Uint8Array) => {
+        const src = _uncheckedConverter.littleEndianToNewArray(num, dest.length)
+        dest.set(src)
+      },
+      arrayToLittleEndian: module.toBigIntLE as UncheckedConverter['arrayToLittleEndian'],
+      arrayToBigEndian: module.toBigIntBE as UncheckedConverter['arrayToBigEndian']
+    }
+  }
+} catch (e) {
+  // bigint-buffer not available as peer-dependency
 }
```

Comment:
You may want to consider narrowing the scope of this try/catch block to just the import.  It is possible that the import succeeds but then some other error occurs between lines 9 and 23, in which case that error would be swallowed by the catch.

Reply:
If another error occurs (the only way for that piece of code to error is if bigint-buffer switches to proxies) i still want the fallback implementation to go through.